### ISSUE TITLE
Arabic letters unicode issue

### ISSUE THREAD

**Issue by jihadbourassi** [matthewearl/deep-anpr#107]
Hello everyone
I am trying to implement this code for moroccan licence plate detection, the moroccan number plates contain arabic letters and number and i am having trouble generating the dataset of images

**Comment by jihadbourassi** [comment]
UPDATE : I ended up making gen.py work and generated the dataset with the arabic letters using UTF-8 encodings for CHARS but now when i tried launching training i was getting encoding errors (i bypassed them by redefining CHARS as a non encoded string) but now i am getting incompatible tensor shape errors. I am new with Tensorflow can someone help me ?

**Comment by ghost** [comment]
duuuuude u making this for moroccan licence plates ???

**Comment by jihadbourassi** [comment]
@ILYASMOUTAWWAKIL  
Yes i am, i was able to adapt the code, genrate the dataset and launch the training, i'm waiting on the loss function to converge it might take a while since i don't have gpu.

I posted the code [here](https://github.com/jihadbourassi/deep-anpr-for-Moroccan-licence-plates)

**Comment by ghost** [comment]
And what abt real time detection the code mathew uses is quite hard coding
the plate detection .. it takes him a lot of time to process one image
while using his GPU .. his solution is not practical for embedded systems ..

**Comment by jihadbourassi** [comment]
@ILYASMOUTAWWAKIL  Since my academic project is about processing one image at a time and not live detection in a video, I'll be satisfied if I achieve just that. But you're right, I'm aware that Matthew's code is not ideal for real time recognition, I will definitely expand upon it once I am done with my project obligations.

**Comment by zongoalbert** [comment]
@jihadbourassi  could you share with us the file : weights.npz
